timesketch/lib/utils.py

```diff
@@ -246,12 +246,46 @@ def read_and_validate_jsonl(file_handle):
     """
     # Fields that must be present in each entry of the JSONL file.
     mandatory_fields = ['message', 'datetime', 'timestamp_desc']
-    lineno = 0
+    line_number = 0
+    max_len_line_jsonl = current_app.config.get('MAX_LENGTH_LINE_JSONL')
+    tag_reducted = current_app.config.get('TAG_REDUCTED_JSONL')
     for line in file_handle:
-        lineno += 1
+        line_number += 1
         try:
             linedict = json.loads(line)
             ld_keys = linedict.keys()
+
+            current_len = len(line)
+            if current_len > max_len_line_jsonl:
+                field_len={}
+                total_len=0
+                for x in ld_keys:
+                    if isinstance(linedict[x], str):
+                        field_len[x]=len(linedict[x])
+                        total_len+=field_len[x]
+                for x in ld_keys:
+                    if field_len[x] > max_len_line_jsonl:
+                        if (total_len-field_len[x]) > max_len_line_jsonl:
```

Comment:
ok, let's imagine this... total length possible is 50 characters, and you've got a line that is 80 characters long, ok, so it is too long, but you go over each of the fields and all of them are just 40 characters long (2 columns in the line), so you'll have here
```
if (80 - 40) > 50:
```
so `if 40 > 50`, all of these will not trigger, however the total length is still longer than the maximum allowed

the question is how is it best to handle that?

a) you can do something `advanced` or you can start by doing the simple thing

1. check if length of any field is greater than the max, and then reduce it by the amount needed to have the total line length below the threshold
  --> this  would not work in the above example, so you'll move on

2. go through the fields and as soon as you start to get close to the limit you just cut all of the remaining fields shorter...
    --> first column is 40 chars, and then the second one is only 7 chars plus '...' to indicate it was cut short

3. or you calculate what fields are the longest and then you cut each one of them by enough characters to fit 
    --> This is the alternative, you can have then field A as 22 chars + '...' and then field B as 22 chars and '...', combining total to 50, but still keeping a bit more of field B than in the other approach

Reply:
Ok thanks for your idea, i will work it